pkgs/servers/uwsgi/default.nix

```diff
@@ -83,6 +89,10 @@ stdenv.mkDerivation rec {
     substituteAll ${./nixos.ini} buildconf/nixos.ini
   '';
 
+  # this is a hack to make the php plugin link with session.so (which on nixos is a separate package)
+  # the hack works in coordination with ./0002-additional-php-ldflags.patch
+  UWSGICONFIG_PHP_LDFLAGS = "-Wl,-rpath=${php-embed.extensions.session}/lib/php/extensions/,--library-path=${php-embed.extensions.session}/lib/php/extensions/,-l:session.so";
```

Comment:
Does uwsgi still build when the php plugin is not enabled? Doesn't this require php-embed to be built on all systems that install uwsgi?
The patches should be applied conditionally when the PHP plugin is enabled so the uwsgi build doesn't break when php-embed breaks.

Reply:
good point. let me double check this.
thanks for reviewing.

Reply:
- uwsgi builds even without the php plugin enabled
- the patches are just to the parts of the uwsgi build system for the php plugin itself, so they don't change the uwsgi build process at all
- you are correct that setting UWSGICONFIG_PHP_LDFLAGS unconditionally pulls php-embed as a dependency to uwsgi even if php is not used. I'll fix this.

Reply:
fixed in b55cc92f5cc81d47e68330b85d71ce7422b175a6